docs/graph-algorithms/breadth-first-search.mdx

```diff
@@ -0,0 +1,101 @@
+---
+title: Breadth First Search
+slug: /graph-algorithms/breadth-first-search
+---
+
+## Introduction
+Breadth First Search Algorithm (BFS) is a Popular algorith to find shortest path in unwighted graph
+
+## Explanation
+
+- Given an undirected & Connected Graph (V, E)
+  V: Vertices E: Edges
+- Goal: methodically explore every vertex and every edge
+- “Explore” a graph, turning it into a tree
+- One vertex at a time
+- Expand frontier of explored vertices across its breadth
+- Discovers all vertices at distance k from s before those at distance k+1
+- A level-by-level order traversal
+
+##APPS
+
+- Find Connected Components in Undirected G
+- Web Crawling
+- Social N/W: Friend Finder
+- Pocket Cube (2x2x2 Rubic Cube)
+- Garbage Collector
+
+
+## Algorithm
+
+- Builds a tree over the graph
+- Pick a source vertex to be the root
+- Find (“discover”) all of its children, then all of their children, etc.
+
+- Associate vertex “colors” to guide the algorithm
+- white  vertices have not been discovered
+- All vertices start out white
+- Grey vertices are discovered but not fully explored
+- They may be adjacent to white vertices
+- Black vertices are discovered and fully explored
+- They are adjacent only to black and gray vertices
+- Explore vertices by scanning neighbors of grey vertices
+
+## Code
+
+```cpp
```

Comment:
Add more implementations. See some merged PRs to see how Code is actually added here

Reply:
@HarshCasper Changes are done